rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -141,14 +152,40 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
   }
 
   template<typename T>
-  typename std::enable_if<std::is_same<T, rcl_serialized_message_t>::value, void>::type
+  typename std::enable_if<
+    std::is_same<T, rcl_serialized_message_t>::value &&
+    !std::is_same<MessageT, rclcpp::experimental::SerializedContainer>::value,
```

Comment:
again, I don't understand why we need that check.

Reply:
EDIT: Help me to get this right here.

Is the idea to separate between the four cases
* publish(MsgT) -> subscribe(MsgT)
* publish(MsgT) -> subscribe(rcl_serialized_message_t)
* publish(rcl_serialized_msg_t) -> subscribe(MsgT)
* publish(rcl_serialized_msg_t) -> subscribe(rcl_serialized_msg_t)

where this precise specialization is `publish(rcl_serialized_msg_t) -> subscribe(MsgT)` ?

Reply:
I have added comments for clarification to the code